src/GafferOSL/ShadingEngine.cpp

```diff
@@ -431,13 +526,158 @@ struct ThreadRenderState
 namespace
 {
 
+template< int WidthT >
+class GafferBatchedRendererServices : public OSL::BatchedRendererServices<WidthT>
+{
+	// To avoid having the specify the WidthT parameter
+	// to every templated wrapper, we can create class
+	// scoped aliases for this class's WidthT
+	OSL_USING_DATA_WIDTH(WidthT);
+
+	public :
+
+		GafferBatchedRendererServices<WidthT>( OSL::TextureSystem *textureSystem )
+			:	OSL::BatchedRendererServices<WidthT>( textureSystem )
+		{
+		}
+
+		bool is_overridden_get_inverse_matrix_WmWxWf() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_get_matrix_WmWsWf() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_get_inverse_matrix_WmsWf() const override
+		{
+			return true;
+		}
+
+		bool is_overridden_get_inverse_matrix_WmWsWf() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_texture() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_texture3d() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_environment() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_pointcloud_search() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_pointcloud_get() const override
+		{
+			return false;
+		}
+
+		bool is_overridden_pointcloud_write() const override
+		{
+			return false;
+		}
+
+		Mask get_matrix( BatchedShaderGlobals *sg, Masked<OSL::Matrix44> result, ustring from, Wide<const float> time ) override
```

Comment:
I found this stuff pretty confusing, but I think the basic summary is: all of the is_overridden methods you have to implement ( you can't miss them because they are pure virtual ) are for methods which have a reasonable fallback.  If you don't implement them, you get a default implementation, which does something like call the version with a scalar `from` for each distinct value of `from` in the wide from, which is probably good enough for us, at least for now.

So I think that explains both why there is no is_overridden for the base function that we have to implement, and why it's OK not to override the function variants that we optionall can.  Though this is all kinda complicated looking.